sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,240 @@
+    >>> est.inverse_transform(Xt)
+    array([[-1.5,  1.5, -3.5, -0.5],
+           [-0.5,  2.5, -2.5, -0.5],
+           [ 0.5,  3.5, -1.5,  0.5],
+           [ 0.5,  3.5, -1.5,  1.5]])
+    """
+
+    def __init__(self, n_bins=2, ignored_features=None):
+        self.n_bins = n_bins
+        self.ignored_features = ignored_features
+
+    def fit(self, X, y=None):
+        """Fits the estimator.
+
+        Parameters
+        ----------
+        X : numeric array-like, shape (n_samples, n_features)
+            Data to be discretized.
+
+        y : ignored
+
+        Returns
+        -------
+        self
+        """
+        X = check_array(X, dtype=FLOAT_DTYPES)
+
+        n_features = X.shape[1]
+        ignored = self._check_ignored_features(self.ignored_features,
+                                               n_features)
+        self.transformed_features_ = np.delete(np.arange(n_features), ignored)
+
+        min = np.min(X, axis=0)
+        min[ignored] = 0
+        self.offset_ = min
+
+        n_bins = self._check_n_bins(self.n_bins, n_features, ignored)
+        n_bins[ignored] = 0
+        self.n_bins_ = n_bins
+
+        ptp = np.ptp(X, axis=0)
+        same_min_max = np.where(ptp == 0)[0]
+        if len(same_min_max) > 0:
+            warnings.warn("Fitted X contained constant features at indices "
+                          "{}. All of these features will be discretized to "
+                          "the value 0."
+                          .format(", ".join(str(i) for i in same_min_max)))
+
+        with np.errstate(divide='ignore', invalid='ignore'):
+            bin_widths = ptp / self.n_bins_
+        bin_widths[ignored] = 0
+        self.bin_width_ = bin_widths
+        return self
+
+    @staticmethod
+    def _check_n_bins(n_bins, n_features, ignored):
+        if isinstance(n_bins, numbers.Number):
+            if n_bins < 2:
+                raise ValueError("Discretizer received an invalid number "
+                                 "of bins. Received {0}, expected at least 2."
+                                 .format(n_bins))
+            return np.ones(n_features) * n_bins
+
+        n_bins = check_array(n_bins, dtype=int, copy=True, ensure_2d=False)
+
+        if n_bins.ndim > 1 or n_bins.shape[0] != n_features:
+            raise ValueError("n_bins must be a scalar or array "
+                             "of shape (n_features,).")
+
+        bad_nbins_value = n_bins < 2
+        is_ignored = np.zeros(n_features).astype(bool)
+        is_ignored[ignored] = True
+
+        violating_indices = np.where(~is_ignored & bad_nbins_value)[0]
+        if violating_indices.shape[0] > 0:
+            indices = ", ".join(str(i) for i in violating_indices)
+            raise ValueError("Discretizer received an invalid number "
+                             "of bins at indices {}. Number of bins "
+                             "must be at least 2."
+                             .format(indices))
```

Comment:
Looking into this, if I did

```
raise ValueError("Discretizer received an invalid number "
                            "of bins at indices {}. Number of bins "
                            "must be at least 2."
                            .format(indices.tolist()))
```

This would format to be

```
Discretizer received an invalid number of bins at indices [1, 2, 3]. Number of bins must be at least 2.
```

instead of

```
Discretizer received an invalid number of bins at indices 1, 2, 3. Number of bins must be at least 2.
```

I personally liked the second formatted string better.